Package.swift

```diff
@@ -35,7 +35,6 @@ let package = Package(
         .macOS(.v10_15), .iOS(.v12), .tvOS(.v12)
     ],
     products: [
```

Comment:
Don't we want to remove this root swift file completely?

Reply:
Yes, we will. At that point, no one would be able to target this repo with SPM. For this PR, I'm only separating out AzureCore to see what that would look like.

Reply:
OK once we do move over I expect folks can still use an older release tag for the preview packages we did ship from the repo still.

Reply:
Yes, the old release tags will work

Reply:
Removed the root Package.swift.